scene/resources/particles_material.cpp

```diff
@@ -554,8 +554,8 @@ void ParticlesMaterial::_update_shader() {
 		}
 	}
 	//scale by scale
-	code += "	float base_scale = mix(scale * tex_scale, 1.0, scale_random * scale_rand);\n";
-	code += "	if (base_scale == 0.0) {\n";
+	code += "	float base_scale = sqrt(tex_scale) * mix(scale * sqrt(tex_scale), 1.0, scale_random * scale_rand);\n";
```

Comment:
What does this change achieve/why does it fix the issue? It will increase computational cost to do `sqrt(tex_scale)` twice instead of using it directly.

Reply:
If `float base_scale = mix(scale * tex_scale, 1.0, scale_random * scale_rand);`
<=> `base_scale = scale * tex_scale*(1-scale_random * scale_rand) + 1.0*scale_random * scale_rand`
Suppose tex_scale is a function of time with form `f(t) = text_scale = a*t +b` (a, b is constant through lifetime of particle)
=> derivative of `text_scale = a`
=> derivative of  `base_scale = scale*a*(1-scale_random * scale_rand)`. This is a constant through lifetime of particle. This is lead to weird behavior of particles because particle with large scale will disappear before particle with small scale (large scale will have large speed of change).

Therefore we have to make derivative of  base_scale  change through lifetime. I have tried many functions and sqrt() function have a good result.

Reply:
You might want to pull that sqrt call into a seperate variable so you don't call it twice here as sqrt can be expensive.
The compiler would probably optimize it away but it might also not do it so thats my 2 cents here.

Reply:
A proper solution is actually just to move ``tex_scale`` outside the ``mix()`` function. 

``float base_scale = tex_scale * mix(scale, 1.0, scale_random * scale_rand);``